### PR TITLE
Enable running gprofiler executable on Alpine

### PR DESCRIPTION
**Pull request by mpozniak95**
## Description
Alpine grep doesn't support flag -P (Perl compatible regular expressions), so pgrep_map method now uses -E (Extended regular expressions)

## Related Issue
https://github.com/Granulate/gprofiler/issues/636

### CODE REVIEW

File: gprofiler/profilers/python.py [hunk 1]
```diff
@@ -249,7 +249,9 @@ def _select_processes_to_profile(self) -> List[Process]:
         if is_windows():
             all_processes = [x for x in pgrep_exe("python")]
         else:
-            all_processes = [x for x in pgrep_maps(DETECTED_PYTHON_PROCESSES_REGEX)]
+            all_processes = [
+                x for x in pgrep_maps(r"(^.+/(lib)?python[^/]*$)|(^.+/site-packages/.+?$)|(^.+/dist-packages/.+?$)")
```
[the remark this answers is not in the document]
Comment (mpozniak95) [reply gprofiler/profilers/python.py:253]: Wasn't sure that if I change it in granulate-utils if this won't change something in other repository.

[the remark this answers is not in the document]
Comment (mpozniak95) [reply gprofiler/profilers/python.py:253]: Added PR in granulate-utils: https://github.com/Granulate/granulate-utils/pull/124

[the remark this answers is not in the document]
Comment (Jongy) [reply gprofiler/profilers/python.py:253]: Any other pattern we should fix?

[the remark this answers is not in the document]
Comment (mpozniak95) [reply gprofiler/profilers/python.py:253]: We use pgrep_maps in four places in code, I changed three patterns and java pattern works with grep -E flag too

[the remark this answers is not in the document]
Comment (Jongy) [reply gprofiler/profilers/python.py:253]: Gotcha

Comment (Jongy) [inline gprofiler/profilers/python.py:253]: I merged - so update the granulate-utils reference to https://github.com/Granulate/granulate-utils/commit/cb66a9fe803d0448ac6eea6aceb85e34a88b0b8f, and revert this change

File: gprofiler/utils/__init__.py [hunk 2]
```diff
@@ -314,7 +314,7 @@ def pgrep_exe(match: str) -> List[Process]:
 def pgrep_maps(match: str) -> List[Process]:
     # this is much faster than iterating over processes' maps with psutil.
     result = run_process(
-        f"grep -lP '{match}' /proc/*/maps",
+        f"grep -lE '{match}' /proc/*/maps",
```
Comment (Jongy) [inline gprofiler/utils/__init__.py:317]: You can comment here explaining about the choice of `-E` over `-P` (to support systems where grep is not PCRE)

**Review by Jongy — CHANGES_REQUESTED** [review]
Final comments

**Review by Jongy — APPROVED** [review]

**Comment by Jongy** [comment]
Good job :)

**Pull request merged**